File: Climbingstairs.py
class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        dp = [1,1]
        for i in range(1,n):
            a = dp[0]+dp[1]
            dp[0] = dp[1]
            dp[1] = a
        return dp[1]    

File: Javascript/Climbing Stairs.js
/**
 * @param {number} n
 * @return {number}
 */

// Dp solution for climbing problem
var climbStairs = function(n) {
    
    var a = 1;
    var b = 1;
    for(i = 1;i<n;i++){
        var temp = b
        b = a + b;
        a = temp
    }
    return b
};
